src/extra/AutoUI/CustomWidget.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import {
+	Format,
+	JsonTypes,
+	UiSchema,
+	Value,
+} from '../../components/Renderer/types';
+import { JSONSchema7 as JSONSchema } from 'json-schema';
+import { transformUiSchema } from '../../components/Renderer/widgets/widget-util';
+import { getValue, getWidget } from '../../components/Renderer';
+import castArray from 'lodash/castArray';
+import { AutoUIBaseResource, AutoUIContext, Priorities } from './schemaOps';
+import { TableColumn } from '../../components/Table';
+import { getSelected, getSortingFunction } from './utils';
+
+interface CustomWidgetProps {
+	value: Value;
+	extraContext: object | undefined;
+	schema: JSONSchema;
+	extraFormats: Format[];
+	uiSchema?: UiSchema;
+}
+
+export const CustomWidget = ({
+	value,
+	extraContext,
+	schema,
+	extraFormats,
+	uiSchema,
+}: CustomWidgetProps) => {
+	const processedUiSchema = transformUiSchema({
+		value,
+		uiSchema,
+		extraContext,
+	});
+
+	const processedValue = getValue(value, schema, processedUiSchema);
+	const types = schema?.type != null ? castArray(schema.type) : [];
+
+	if (
+		processedValue === undefined ||
+		(processedValue === null && !types.includes(JsonTypes.null))
+	) {
+		return null;
+	}
+
+	const Widget = getWidget(
+		processedValue,
+		schema.format,
+		undefined,
+		extraFormats,
+	);
+
+	return (
+		<Widget
+			extraContext={extraContext}
+			extraFormats={extraFormats}
+			value={processedValue}
+			schema={schema}
+		/>
+	);
+};
+
+export const getColumnsFromSchema = <T extends AutoUIBaseResource<T>>({
```

Comment:
Understandable, I put it here because it was rendering CustomWidgets. By name it sounds like it fits better inside schemaOps but it does not work in there (not exactly sure why). I've put it inside AutoUI's index for now, if you kbow why it may not be working inside schemaOps or if you can think of a better place to put it, please let me know